### ISSUE TITLE
Incremental training on a dataset with added features is broken since r0.17.3. 

### ISSUE THREAD

**Issue by andrey-khropov** [catboost/catboost#1196]
Problem: subj
catboost version: 0.17.3+
Operating System: Ubuntu 16.04
CPU: 2 x Intel(R) Xeon(R) CPU E5-2660 0 @ 2.20GHz

```py
from catboost import CatBoostRegressor

# initial training

train_data_1 = [
    [1, 4, 5, 6],
    [4, 5, 6, 7],
    [30, 40, 50, 60]
]

train_labels_1 = [10, 20, 30]

model1 = CatBoostRegressor(iterations=2)

model1.fit(train_data_1, train_labels_1)

# continue training with a different dataset

train_data_2 = [
    [8, 9, 10, 11, 12, 13],
    [70, 80, 90, 100, 110, 120]
]

train_labels_2 = [40, 50]

model2 = CatBoostRegressor(iterations=2)

model2.fit(train_data_2, train_labels_2,  init_model=model1)

```

leads to segmentation fault

**Comment by andrey-khropov** [comment]
This bug has been fixed in https://github.com/catboost/catboost/commit/6959d382d44f140d1824eb6679622471f4f1202e and will be included in the next release.

**Comment by H-abbas-10** [comment]
> This bug has been fixed in [6959d38](https://github.com/catboost/catboost/commit/6959d382d44f140d1824eb6679622471f4f1202e) and will be included in the next release.

i seem to have a bug where there i get an index error. EG
 
i train my model with 10 features.

 I then train another catboost model with 12 features (same 10 features as before + 2 extra ones) but i reference the first model using init_model paramater. Not sure how to get around this index error?
